pulse2percept/electrode2currentmap.py

```diff
@@ -570,6 +597,121 @@ def get_pulse(pulse_dur, tsample, interphase_dur, pulsetype):
     return pulse
 
 
+class Image2Pulsetrain(TimeSeries):
+    """Converts an image into a series of pulse trains
+
+    This function creates an input stimulus from an RGB or grayscale image.
+    Requires `scikit-image`.
+
+    Parameters
+    ----------
+    img : str|array_like
+        An input image, either a valid filename (string) or a numpy array
+        (row x col x channels).
+    implant : ElectrodeArray
+        An ElectrodeArray object that describes the implant.
+    coding : {'amplitude', 'frequency'}, optional
+        A string describing the coding scheme:
+        - 'amplitude': Image intensity is linearly converted to a current
+                       amplitude between `valrange[0]` and `valrange[1]`.
+                       Frequency is held constant at `const_freq`.
+        - 'frequency': Image intensity is linearly converted to a pulse
+                       frequency between `valrange[0]` and `valrange[1]`.
+                       Amplitude is held constant at `const_amp`.
+        Default: 'amplitude'
+    valrange : list, optional
+    const_amp : float, optional
+    const_freq : float, optional
+    rftype : {'square', 'gaussian'}, optional
+    rfsize : float, optional
+    tsample : float, optional
+        Sampling time step (seconds). Default: 0.005 / 1000 seconds.
+    dur : float, optional
+        Stimulus duration (seconds). Default: 0.5 seconds.
+    pulsedur : float, optional
+    interphasedur : float, optional
+    pulsetype: {'cathodicfirst', 'anodicfirst'}, optional
+
+    """
+
+    # TODO: Default RF size could be half of the electrode gap?
+    def __init__(self, img, implant, coding='amplitude', valrange=[0, 50],
+                 const_amp=20, const_freq=20, rftype='gaussian', rfsize=None,
+                 tsample=0.005 / 1000, dur=0.5, pulsedur=0.5 / 1000.,
+                 interphasedur=0.5 / 1000., pulsetype='cathodicfirst'):
+        try:
+            from skimage.io import imread
+            from skimage.transform import resize
+            from skimage.color import rgb2gray
+        except ImportError:
+            raise ImportError("You do not have scikit-image installed.")
+
+        # Make sure range of values is valid
+        assert len(valrange) == 2 and valrange[1] > valrange[0]
+
+        if isinstance(img, str):
+            # Load image from filename
+            img_large = imread(img, as_grey=True).astype(np.float32)
+        else:
+            # Convert image to grayscale
+            img_large = rgb2gray(np.array(img)).astype(np.float32)
+
+        # Maximize contrast
```

Comment:
Wouldn't this lead to confusing results in an experiment in which different contrasts are used?